src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py

```diff
@@ -10153,3 +10153,570 @@ def test_aks_upgrade_upgrade_settings(self, resource_group, resource_group_locat
+            create_keyvault_cmd,
+            checks=[
+                self.check("properties.provisioningState", "Succeeded"),
+                self.check("properties.enableRbacAuthorization", True),
+                self.check("name", kv_name),
+            ],
+        ).get_output_in_json()
+        keyvault_id = keyvault["id"]
+
+        self.kwargs.update({"keyvault_id": keyvault_id})
+
+        # create cluster with app routing enabled
+        create_cmd = (
+            "aks create --resource-group={resource_group} --name={aks_name} --location={location} "
+            "--ssh-key-value={ssh_key_value} --enable-app-routing"
+        )
+        result = self.cmd(
+            create_cmd,
+            checks=[
+                self.check("provisioningState", "Succeeded"),
+                self.check("ingressProfile.webAppRouting.enabled", True),
+            ],
+        ).get_output_in_json()
+        object_id = result["ingressProfile"]["webAppRouting"]["identity"]["objectId"]
+        self.kwargs.update({"object_id": object_id})
+
+        # update with enable_rbac_authorization flag in keyvault set to true
+        update_cmd = (
+            "aks approuting update --resource-group={resource_group} --name={aks_name} --enable-kv "
+            "--attach-kv {keyvault_id}"
+        )
+
+        self.cmd(
+            update_cmd,
+            checks=[
+                self.check("provisioningState", "Succeeded"),
+                self.check("ingressProfile.webAppRouting.enabled", True),
+                self.check("addonProfiles.azureKeyvaultSecretsProvider.enabled", True),
+            ],
+        )
+
+        # update keyvault with rbac auth disabled
+        update_keyvault_cmd = "keyvault update --resource-group={resource_group} --name={kv_name} --enable-rbac-authorization=false"
+        self.cmd(
+            update_keyvault_cmd,
+            checks=[
+                self.check("properties.provisioningState", "Succeeded"),
+                self.check("properties.enableRbacAuthorization", False),
+                self.check("name", kv_name),
+            ],
+        )
+
+        # update with enable_rbac_authorization flag in keyvault set to false
+        update_cmd = (
+            "aks approuting update --resource-group={resource_group} --name={aks_name} "
+            "--attach-kv {keyvault_id}"
+        )
+
+        self.cmd(
+            update_cmd,
+            checks=[
+                self.check("provisioningState", "Succeeded"),
+                self.check("ingressProfile.webAppRouting.enabled", True),
+                self.check("addonProfiles.azureKeyvaultSecretsProvider.enabled", True),
+            ],
+        )
+
+        check_access_policy_cmd = "az keyvault show --resource-group={resource_group} --name={kv_name} --query \"properties.accessPolicies[?objectId=='{object_id}']\" -o json"
+        self.cmd(
+            check_access_policy_cmd,
+            checks=[
+                self.check("length(@)", 1),
```

Comment:
something wrong with this. this is what's failing in tests

azure.cli.testsdk.exceptions.JMESPathCheckAssertionError: Query 'length(@)' doesn't yield expected value '1', instead the actual value is 'None'. Data:

Reply:
fixed with kv updates